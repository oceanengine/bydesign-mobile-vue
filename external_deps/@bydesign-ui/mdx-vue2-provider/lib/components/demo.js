/**
 * Component Demo
 */
import createBEM from '../utils/bem';
import HL from './highlight';
import { openCodepen, openSandbox, copyToClipboard } from '../utils/editor';
var bem = createBEM('demo');

var Demo = function Demo() {
  return {
    props: {
      title: [String, Number],
      desc: [String, Number],
      code: String,
      src: String,
      srcRender: Function,
      maxHeight: Number
    },
    data: function data() {
      return {
        unfolded: false
      };
    },
    methods: {
      setUnfolded: function setUnfolded(value) {
        this.unfolded = value;
      }
    },
    render: function render() {
      var h = arguments[0];
      var title = this.title,
          desc = this.desc,
          code = this.code,
          srcRender = this.srcRender,
          unfolded = this.unfolded,
          setUnfolded = this.setUnfolded,
          _this$maxHeight = this.maxHeight,
          maxHeight = _this$maxHeight === void 0 ? 200 : _this$maxHeight;
      var containerStyle = {
        maxHeight: "".concat(maxHeight, "px")
      };
      return h("div", {
        "class": bem()
      }, [h("div", {
        "class": bem('title'),
        "attrs": {
          "id": title.toString()
        }
      }, [title]), srcRender !== undefined && h("div", {
        "class": bem('box')
      }, [h("div", {
        "class": bem('show')
      }, [srcRender()]), desc && h("div", {
        "class": bem('desc')
      }, [desc]), code !== undefined && h("div", {
        "class": bem('code-tool')
      }, [h("div", {
        "class": bem('operations')
      }, [h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openCodepen(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/codepen.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openSandbox(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/sandbox.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            copyToClipboard(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/copy.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            setUnfolded(!unfolded);
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/code.svg')
        }
      })])]), code !== undefined && unfolded && h("div", {
        "class": bem('code'),
        "style": containerStyle
      }, [h(HL, {
        "attrs": {
          "code": unescape(code)
        }
      })])])]);
    }
  };
};

export default Demo;