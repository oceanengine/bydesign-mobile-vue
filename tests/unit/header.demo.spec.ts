import Demo from '@src/components/header/demo/index.vue';
import { snapshotDemo } from '../demo';

snapshotDemo(Demo);
