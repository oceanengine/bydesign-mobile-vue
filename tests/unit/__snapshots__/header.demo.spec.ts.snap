// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div class="demo checker">
  <div class="byted-header--wrapper">
    <div class="byted-header byted-header--default byted-header--fixed byted-hairline--bottom">
      <div class="byted-header__left"><i class="byted-header__icon" style="width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title"><span class="demo-header-icon"><img src="https://s2.pstatp.com/cdn/expire-1-d/byted-creative-app/static/ByDesign-logo.svg" class="img-icon"></span></div>
    </div>
  </div>
  <div></div>
  <div>
    <div class="byted-header byted-header--default byted-hairline--bottom">
      <div class="byted-header__left"><i class="byted-header__icon" style="width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title">ByDesign</div>
    </div>
  </div>
  <div>
    <div class="byted-header byted-header--default byted-hairline--bottom">
      <div class="byted-header__title">ByDesign</div>
    </div>
  </div>
  <div>
    <div class="byted-header byted-header--default byted-hairline--bottom">
      <div class="byted-header__left">
        <div class="byted-button byted-button--text byted-button--normal">取消</div>
      </div>
      <div class="byted-header__title">ByDesign</div>
      <div class="byted-header__right">
        <div class="byted-button byted-button--text byted-button--normal">确定</div>
      </div>
    </div>
  </div>
  <div>
    <div class="byted-header byted-header--primary">
      <div class="byted-header__left"><i class="byted-header__icon" style="fill: #ffffff; width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title">ByDesign</div>
    </div>
  </div>
  <div>
    <div class="byted-header byted-header--default byted-hairline--bottom" style="margin-bottom: 20px;">
      <div class="byted-header__left"><i class="byted-header__icon" style="width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title">
        <div class="byted-header-tabs byted-header-tabs--default">
          <div class="byted-header-tabs__box">
            <div class="byted-header-tab byted-header-tab--default byted-header-tab--active">
              <div class="byted-header-tab__box">字节跳动</div>
            </div>
            <div class="byted-header-tab byted-header-tab--default">
              <div class="byted-header-tab__box"> 巨量引擎 </div>
            </div>
          </div>
          <div class="byted-header-tabs__line"></div>
        </div>
      </div>
    </div>
    <div class="byted-header byted-header--primary">
      <div class="byted-header__left"><i class="byted-header__icon" style="fill: #ffffff; width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title">
        <div class="byted-header-tabs byted-header-tabs--primary">
          <div class="byted-header-tabs__box">
            <div class="byted-header-tab byted-header-tab--primary byted-header-tab--active">
              <div class="byted-header-tab__box">字节跳动</div>
            </div>
            <div class="byted-header-tab byted-header-tab--primary">
              <div class="byted-header-tab__box"> 巨量引擎 </div>
            </div>
          </div>
          <div class="byted-header-tabs__line"></div>
        </div>
      </div>
    </div>
  </div>
  <div class="demo-cell"></div>
</div>
`;
